src/topology/algebra/floor_ring.lean

```diff
@@ -0,0 +1,225 @@
+  rw ceil_eq_iff,
+  exact ⟨sub_one_lt _, le_refl _⟩
+end
+
+lemma tendsto_floor_left [order_closed_topology α] (n : ℤ) :
+  filter.tendsto (λ x, floor x : α → α) (𝓝[Iio n] n) (𝓝[Iic (n-1)] (n-1)) :=
+begin
+  rw ← nhds_within_Ico_eq_nhds_within_Iio (sub_one_lt (n : α)),
+  convert (tendsto_nhds_within_congr $ floor_eq_on_Ico (n-1))
+    (tendsto_nhds_within_of_tendsto_nhds_of_eventually_within _ tendsto_const_nhds
+      (filter.eventually_of_forall (λ _, mem_Iic.mpr $ le_refl _)));
+  norm_cast <|> apply_instance,
+  ring
+end
+
+lemma tendsto_ceil_right [order_closed_topology α] (n : ℤ) :
+  filter.tendsto (λ x, ceil x : α → α) (𝓝[Ioi n] n) (𝓝[Ici (n+1)] (n+1)) :=
+begin
+  rw ← nhds_within_Ioc_eq_nhds_within_Ioi (lt_add_one (n : α)),
+  convert (tendsto_nhds_within_congr $ ceil_eq_on_Ioc (n+1))
+    (tendsto_nhds_within_of_tendsto_nhds_of_eventually_within _ tendsto_const_nhds
+      (filter.eventually_of_forall (λ _, mem_Ici.mpr $ le_refl _)));
+  norm_cast <|> apply_instance,
+  ring
+end
+
+lemma tendsto_floor_left' [order_closed_topology α] (n : ℤ) :
+  filter.tendsto (λ x, floor x : α → α) (𝓝[Iio n] n) (𝓝 (n-1)) :=
+begin
+  rw ← nhds_within_univ,
+  exact tendsto_nhds_within_mono_right (subset_univ _) (tendsto_floor_left n),
+end
+
+lemma tendsto_ceil_right' [order_closed_topology α] (n : ℤ) :
+  filter.tendsto (λ x, ceil x : α → α) (𝓝[Ioi n] n) (𝓝 (n+1)) :=
+begin
+  rw ← nhds_within_univ,
+  exact tendsto_nhds_within_mono_right (subset_univ _) (tendsto_ceil_right n),
+end
+
+lemma continuous_on_fract [topological_add_group α] (n : ℤ) :
+  continuous_on (fract : α → α) (Ico n (n+1) : set α) :=
+continuous_on_id.sub (continuous_on_floor n)
+
+lemma tendsto_fract_left' [order_closed_topology α] [topological_add_group α]
+  (n : ℤ) : filter.tendsto (fract : α → α) (𝓝[Iio n] n) (𝓝 1) :=
+begin
+  convert (tendsto_nhds_within_of_tendsto_nhds filter.tendsto_id).sub (tendsto_floor_left' n);
+  [{norm_cast, ring}, apply_instance, apply_instance]
+end
+
+lemma tendsto_fract_left [order_closed_topology α] [topological_add_group α]
+  (n : ℤ) : filter.tendsto (fract : α → α) (𝓝[Iio n] n) (𝓝[Iio 1] 1) :=
+tendsto_nhds_within_of_tendsto_nhds_of_eventually_within _
+  (tendsto_fract_left' _) (filter.eventually_of_forall fract_lt_one)
+
+lemma tendsto_fract_right' [order_closed_topology α] [topological_add_group α]
+  (n : ℤ) : filter.tendsto (fract : α → α) (𝓝[Ici n] n) (𝓝 0) :=
+begin
+  convert (tendsto_nhds_within_of_tendsto_nhds filter.tendsto_id).sub (tendsto_floor_right' n);
+  [exact (sub_self _).symm, apply_instance, apply_instance]
+end
+
+lemma tendsto_fract_right [order_closed_topology α] [topological_add_group α]
+  (n : ℤ) : filter.tendsto (fract : α → α) (𝓝[Ici n] n) (𝓝[Ici 0] 0) :=
+tendsto_nhds_within_of_tendsto_nhds_of_eventually_within _
+  (tendsto_fract_right' _) (filter.eventually_of_forall fract_nonneg)
+
+local notation `I` := (Icc 0 1 : set α)
+
+lemma continuous_on.comp_fract {β : Type*} [order_topology α]
```

Comment:
Did you try to deduce this lemma from the next one (with `unit` as a parameter space)? Or does it require so much plumbing that you don't gain anything?